A part of CACHE_INCREMENT_DYN_STAT can be called unintentionally

gcc 8 on fedora 28 (not released yet) found this:
```
In file included from P_Cache.h:41,
                 from CacheRead.cc:24:
CacheRead.cc: In member function 'int CacheVC::openReadStartEarliest(int, Event*)':
P_CacheInternal.h:170:3: error: macro expands to multiple statements [-Werror=multistatement-macros]
   RecIncrRawStat(cache_rsb, mutex->thread_holding, (int)(x), 1); \
   ^~~~~~~~~~~~~~
CacheRead.cc:957:5: note: in expansion of macro 'CACHE_INCREMENT_DYN_STAT'
     CACHE_INCREMENT_DYN_STAT(cache_read_busy_success_stat);
     ^~~~~~~~~~~~~~~~~~~~~~~~
CacheRead.cc:956:3: note: some parts of macro expansion are not guarded by this 'if' clause
   if (write_vc)
   ^~
```

https://github.com/apache/trafficserver/blob/master/iocore/cache/CacheRead.cc#L956-L957

This is a bug but not a compatibility issue. The `if` statement definitely needs braces (or the macro need to be defined as a block).



I guess this is not the only place that has the bug. I'll update this issue if I found more.

--- a/iocore/cache/CacheRead.cc
+++ b/iocore/cache/CacheRead.cc
@@ -953,8 +953,9 @@ CacheVC::openReadStartEarliest(int /* event ATS_UNUSED */, Event * /* e ATS_UNUS
 Lcallreturn:
   return handleEvent(AIO_EVENT_DONE, nullptr); // hopefully a tail call
 Lsuccess:
-  if (write_vc)
+  if (write_vc) {
     CACHE_INCREMENT_DYN_STAT(cache_read_busy_success_stat);
+  }
   SET_HANDLER(&CacheVC::openReadMain);
   return callcont(CACHE_EVENT_OPEN_READ);
 }

--- a/iocore/cache/P_CacheInternal.h
+++ b/iocore/cache/P_CacheInternal.h
@@ -165,23 +165,31 @@ extern RecRawStatBlock *cache_rsb;
 #define CACHE_SET_DYN_STAT(x, y) \
   RecSetGlobalRawStatSum(cache_rsb, (x), (y)) RecSetGlobalRawStatSum(vol->cache_vol->vol_rsb, (x), (y))
 
-#define CACHE_INCREMENT_DYN_STAT(x)                              \
-  RecIncrRawStat(cache_rsb, mutex->thread_holding, (int)(x), 1); \
-  RecIncrRawStat(vol->cache_vol->vol_rsb, mutex->thread_holding, (int)(x), 1);
+#define CACHE_INCREMENT_DYN_STAT(x)                                              \
+  do {                                                                           \
+    RecIncrRawStat(cache_rsb, mutex->thread_holding, (int)(x), 1);               \
+    RecIncrRawStat(vol->cache_vol->vol_rsb, mutex->thread_holding, (int)(x), 1); \
+  } while (0);
 
-#define CACHE_DECREMENT_DYN_STAT(x)                               \
-  RecIncrRawStat(cache_rsb, mutex->thread_holding, (int)(x), -1); \
-  RecIncrRawStat(vol->cache_vol->vol_rsb, mutex->thread_holding, (int)(x), -1);
+#define CACHE_DECREMENT_DYN_STAT(x)                                               \
+  do {                                                                            \
+    RecIncrRawStat(cache_rsb, mutex->thread_holding, (int)(x), -1);               \
+    RecIncrRawStat(vol->cache_vol->vol_rsb, mutex->thread_holding, (int)(x), -1); \
+  } while (0);
 
 #define CACHE_VOL_SUM_DYN_STAT(x, y) RecIncrRawStat(vol->cache_vol->vol_rsb, mutex->thread_holding, (int)(x), (int64_t)y);
 
-#define CACHE_SUM_DYN_STAT(x, y)                                            \
-  RecIncrRawStat(cache_rsb, mutex->thread_holding, (int)(x), (int64_t)(y)); \
-  RecIncrRawStat(vol->cache_vol->vol_rsb, mutex->thread_holding, (int)(x), (int64_t)(y));
+#define CACHE_SUM_DYN_STAT(x, y)                                                            \
+  do {                                                                                      \
+    RecIncrRawStat(cache_rsb, mutex->thread_holding, (int)(x), (int64_t)(y));               \
+    RecIncrRawStat(vol->cache_vol->vol_rsb, mutex->thread_holding, (int)(x), (int64_t)(y)); \
+  } while (0);
 
-#define CACHE_SUM_DYN_STAT_THREAD(x, y)                              \
-  RecIncrRawStat(cache_rsb, this_ethread(), (int)(x), (int64_t)(y)); \
-  RecIncrRawStat(vol->cache_vol->vol_rsb, this_ethread(), (int)(x), (int64_t)(y));
+#define CACHE_SUM_DYN_STAT_THREAD(x, y)                                              \
+  do {                                                                               \
+    RecIncrRawStat(cache_rsb, this_ethread(), (int)(x), (int64_t)(y));               \
+    RecIncrRawStat(vol->cache_vol->vol_rsb, this_ethread(), (int)(x), (int64_t)(y)); \
+  } while (0);
 
 #define GLOBAL_CACHE_SUM_GLOBAL_DYN_STAT(x, y) RecIncrGlobalRawStatSum(cache_rsb, (x), (y))
 

